src/page/EditDues/index.tsx

```diff
@@ -0,0 +1,323 @@
+    setFalse: closeEditStatusModal,
+  } = useBooleanState(false);
+  const [requiredData, setRequiredData] = useState({
+    year: duesYear,
+    memberId: 0,
+    month: 0,
+    status: null,
+    memo: '',
+  });
+  const [status, setStatus] = useState<Status>('PAID');
+
+  const openSnackBar = useSnackBar();
+
+  const { data: allDues, refetch } = useGetAllDues({ year: duesYear });
+  const [filteredValue, setFilteredValue] = useState(allDues.dues);
+
+  const { data: tracks } = useGetTracks();
+  const postDuesMutation = usePostDues();
+  const putDuesMutation = usePutDues();
+
+  const handleNameChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const searchName = e.target.value;
+    if (searchName === '') {
+      if (trackFilter.every((value) => value)) {
+        setFilteredValue(allDues.dues);
+      } else {
+        setFilteredValue(allDues.dues.filter((row) => trackFilter[tracks.map((track) => track.name).indexOf(row.track.name)]));
+      }
+    }
+    setName(searchName);
+  };
+
+  const handleNameSearchClick = () => {
+    if (filteredValue.some((row) => row.name.includes(name))) {
+      setFilteredValue(allDues.dues.filter((row) => row.name.includes(name)));
+    }
+  };
+
+  const handleNameSearchKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === 'Enter') {
+      handleNameSearchClick();
+    }
+  };
+
+  const handleTrackFilterChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const selectedTrack = e.target.name;
+    const trackIndex = tracks.filter((track) => track.name === selectedTrack)[0].id - 1;
+    setTrackFilter((prevTrack) => {
+      const updatedTrack = [...prevTrack];
+      updatedTrack[trackIndex] = !updatedTrack[trackIndex];
+      setFilteredValue(allDues.dues.filter(
+        (row) => updatedTrack[tracks.map((track) => track.name).indexOf(row.track.name)],
+      ));
+      return updatedTrack;
+    });
+  };
+
+  const handleEditStatusModalOpen = (month: number, memberId: number) => {
+    openEditStatusModal();
+    setRequiredData((prev) => {
+      return {
+        ...prev,
+        year: duesYear,
+        memberId,
+        month,
+      };
+    });
+  };
+
+  const handleChangeMutationStatusClick = () => {
+    // TODO: 이전 status와 수정하려는 status가 같은 경우 API 호출이 필요 없음
+    const prevStatus = allDues.dues.find((row) => row.memberId === requiredData.memberId)?.detail.find((detail) => detail.month === requiredData.month)?.status;
+    if (prevStatus === null && prevStatus !== status) {
+      postDuesMutation.mutate({
+        year: requiredData.year,
+        memberId: requiredData.memberId,
+        month: requiredData.month,
+        status,
+        memo: requiredData.memo,
+      });
+    } else if (prevStatus !== null && prevStatus !== status) {
+      putDuesMutation.mutate({
+        year: requiredData.year,
+        memberId: requiredData.memberId,
+        month: requiredData.month,
+        status,
+        memo: requiredData.memo,
+      });
+    }
```

Comment:
```suggestion
    if (prevStatus !== status) {
      if(prevStatus === null){
        postDuesMutation.mutate({
          year: requiredData.year,
          memberId: requiredData.memberId,
          month: requiredData.month,
          status,
          memo: requiredData.memo,
        });
    } else {
      putDuesMutation.mutate({
        year: requiredData.year,
        memberId: requiredData.memberId,
        month: requiredData.month,
        status,
        memo: requiredData.memo,
      });
    }
```
else if 사용을 최대한 자제하면서 위와같이 작성하면 코드 이해도 쉽지않을까 싶습니다.
다만, if문 depth가 이전보다 증가한다는점에서 이게 더 낫다! 라고는 못할거 같아서 참고만 해주셔도 될거같아요😀

Reply:
이전 코드보다 훨씬 가독성에 좋은 것 같아요 수정했습니다! fbf235a575dab73058376c4d94101ed055576b24